### ISSUE TITLE
Unable to handle error due to duplicate key

### ISSUE THREAD

**Issue by bitinn** [pebble/yieldb#4]
Say I have a unique key on field `url`, then these result in error not catch-able by try/catch in a `co` generator function.

```javascript
function* test() {
	try {
		ps.push(articles.insert({url:'http://example.com/'}));
		ps.push(articles.insert({url:'http://example.com/'}));
	} catch(err) {
		console.log(err);
	}
}
```

```
df-air:leaf-server df$ node --harmony task-setup.js 
/Users/df/git/leaf-server/node_modules/yieldb/node_modules/mongodb/lib/mongodb/connection/base.js:246
        throw message;      
              ^
TypeError: undefined is not a function
    at /Users/df/git/leaf-server/node_modules/yieldb/node_modules/mongodb/lib/mongodb/collection/core.js:118:74
    at /Users/df/git/leaf-server/node_modules/yieldb/node_modules/mongodb/lib/mongodb/db.js:1166:7
    at /Users/df/git/leaf-server/node_modules/yieldb/node_modules/mongodb/lib/mongodb/db.js:1899:9
    at Server.Base._callHandler (/Users/df/git/leaf-server/node_modules/yieldb/node_modules/mongodb/lib/mongodb/connection/base.js:453:41)
    at /Users/df/git/leaf-server/node_modules/yieldb/node_modules/mongodb/lib/mongodb/connection/server.js:481:18
    at MongoReply.parseBody (/Users/df/git/leaf-server/node_modules/yieldb/node_modules/mongodb/lib/mongodb/responses/mongo_reply.js:68:5)
    at null.<anonymous> (/Users/df/git/leaf-server/node_modules/yieldb/node_modules/mongodb/lib/mongodb/connection/server.js:439:20)
    at emit (events.js:107:17)
    at null.<anonymous> (/Users/df/git/leaf-server/node_modules/yieldb/node_modules/mongodb/lib/mongodb/connection/connection_pool.js:201:13)
    at emit (events.js:110:17)
```

**Comment by aheckmann** [comment]
this code is pushing functions onto `ps`. I don't see them yielded? is that example correct?

**Comment by bitinn** [comment]
```

/**
 * task-setup.js
 *
 * A simple test script
 */

var co = require('co');
var connect = require('yieldb').connect;

function* setup() {
	var db = yield connect('mongodb://localhost:27017/testdb');
	var test = db.col('test');
	yield test.index({ a:1 }, { unique:1 });
	var ps = [];
	try {
		ps.push(test.insert({ a:1 }));
		ps.push(test.insert({ a:1 }));
	} catch(err) {
		console.log(err);
	}

	return yield ps;
}

var run = co.wrap(setup);

run().then(function(res) {
	console.log('test complete', res);
}).catch(function(err) {
	console.log('test failed', err.message, err.stack);
});
```

```
df-air:leaf-server df$ node --harmony task-setup.js
/Users/df/git/leaf-server/node_modules/yieldb/node_modules/mongodb/lib/mongodb/connection/base.js:246
        throw message;      
              ^
TypeError: undefined is not a function
    at /Users/df/git/leaf-server/node_modules/yieldb/node_modules/mongodb/lib/mongodb/collection/core.js:118:74
    at /Users/df/git/leaf-server/node_modules/yieldb/node_modules/mongodb/lib/mongodb/db.js:1166:7
    at /Users/df/git/leaf-server/node_modules/yieldb/node_modules/mongodb/lib/mongodb/db.js:1899:9
    at Server.Base._callHandler (/Users/df/git/leaf-server/node_modules/yieldb/node_modules/mongodb/lib/mongodb/connection/base.js:453:41)
    at /Users/df/git/leaf-server/node_modules/yieldb/node_modules/mongodb/lib/mongodb/connection/server.js:481:18
    at MongoReply.parseBody (/Users/df/git/leaf-server/node_modules/yieldb/node_modules/mongodb/lib/mongodb/responses/mongo_reply.js:68:5)
    at null.<anonymous> (/Users/df/git/leaf-server/node_modules/yieldb/node_modules/mongodb/lib/mongodb/connection/server.js:439:20)
    at emit (events.js:107:17)
    at null.<anonymous> (/Users/df/git/leaf-server/node_modules/yieldb/node_modules/mongodb/lib/mongodb/connection/connection_pool.js:201:13)
    at emit (events.js:110:17)
```

**Comment by bitinn** [comment]
expected result

```
> use testdb
switched to db testdb
> db.test.insert({a:1})
WriteResult({
	"nInserted" : 0,
	"writeError" : {
		"code" : 11000,
		"errmsg" : "insertDocument :: caused by :: 11000 E11000 duplicate key error index: testdb.test.$a_1  dup key: { : 1.0 }"
	}
})
```

**Comment by aheckmann** [comment]
tracked it down to a bug in the driver when inserting a single document using the "fullResult" option. can you open a bug on the mongodb driver [here](https://jira.mongodb.org/browse/NODE)?

**Comment by bitinn** [comment]
Sure, I do need some repo code though, hard to justify this is a bug on native driver without that.

PS: if this is a case, does it mean this is still broken? https://jira.mongodb.org/browse/NODE-293

**Comment by bitinn** [comment]
OK I have a repo, commented on original issue, if no one notice I will open another ticket.

**Comment by bitinn** [comment]
https://jira.mongodb.org/browse/NODE-344 - looks like fullResult is pretty broken, unless I am hallucinating...

**Comment by aheckmann** [comment]
fixed in 2.2.0
